src/test/java/com/commercetools/sync/categories/CategoryTypeDiffTest.java

```diff
@@ -0,0 +1,478 @@
+        final String newCategoryCustomTypeId = "newCategoryCustomTypeId";
+        final String newCategoryCustomTypeKey = "newCategoryCustomTypeKey";
+        when(newCategoryTypeReference.getId()).thenReturn(newCategoryCustomTypeId);
+        when(newCategoryTypeReference.getKey()).thenReturn(newCategoryCustomTypeKey);
+
+        // Mock existing CustomFields
+        final CustomFields existingCustomFieldsMock = mock(CustomFields.class);
+        when(existingCustomFieldsMock.getType()).thenReturn(categoryTypeReference);
+
+        // Mock new CustomFieldsDraft
+        final CustomFieldsDraft newCustomFieldsMock = mock(CustomFieldsDraft.class);
+        when(newCustomFieldsMock.getType()).thenReturn(newCategoryTypeReference);
+
+        // Mock TypeService and Category Custom Type key Cache.
+        final TypeService typeServiceMock = mock(TypeServiceImpl.class);
+        when(typeServiceMock.getCachedTypeKeyById(anyString())).thenReturn(categoryCustomTypeKey);
+
+        final List<UpdateAction<Category>> updateActions = buildNonNullCustomFieldsActions(existingCustomFieldsMock,
+                newCustomFieldsMock, typeServiceMock);
+
+        assertThat(updateActions).isNotNull();
+        assertThat(updateActions).hasSize(1);
+        assertThat(updateActions.get(0).getAction()).isEqualTo("setCustomType");
+    }
+
+    @Test
+    public void buildNonNullCustomFieldsActions_WithNullExistingCategoryTypeKey_ShouldBuildUpdateActions() {
+        final Reference<Type> categoryTypeReference = mock(Reference.class);
+        final String categoryCustomTypeInternalId = "categoryCustomTypeId";
+        final String categoryCustomTypeKey = "categoryCustomTypeKey";
+        when(categoryTypeReference.getId()).thenReturn(categoryCustomTypeInternalId);
+        when(categoryTypeReference.getKey()).thenReturn(categoryCustomTypeKey);
+
+        // Mock existing CustomFields
+        final CustomFields existingCustomFieldsMock = mock(CustomFields.class);
+        when(existingCustomFieldsMock.getType()).thenReturn(categoryTypeReference);
+
+        // Mock new CustomFieldsDraft
+        final CustomFieldsDraft newCustomFieldsMock = mock(CustomFieldsDraft.class);
+        when(newCustomFieldsMock.getType()).thenReturn(categoryTypeReference);
+
+        // Mock TypeService and Category Custom Type key Cache.
+        final TypeService typeServiceMock = mock(TypeServiceImpl.class);
+        when(typeServiceMock.getCachedTypeKeyById(anyString())).thenReturn(null);
+
+        final List<UpdateAction<Category>> updateActions = buildNonNullCustomFieldsActions(existingCustomFieldsMock,
+                newCustomFieldsMock, typeServiceMock);
+
+        assertThat(updateActions).isNotNull();
+        assertThat(updateActions).hasSize(1);
+        assertThat(updateActions.get(0).getAction()).isEqualTo("setCustomType");
+    }
+
+    @Test
+    public void buildNonNullCustomFieldsActions_WithNullNewCategoryTypeKey_ShouldBuildUpdateActions() {
```

Comment:
do you mean the `type` is absent on the new customFields or the old one? Both are available. But which one do you mean?